src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIDockerService.java

```diff
@@ -0,0 +1,74 @@
+package de.tum.in.www1.artemis.service.connectors.localci;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.PullImageResultCallback;
+import com.github.dockerjava.api.exception.BadRequestException;
+import com.github.dockerjava.api.exception.NotFoundException;
+
+import de.tum.in.www1.artemis.exception.LocalCIException;
+
+/**
+ * Service for Docker related operations in local CI
+ */
+
+@Service
+@Profile("localci")
+
+public class LocalCIDockerService {
```

Comment:
Why did you add these extra empty lines here? In Artemis we normally don't add them.
```suggestion
 */
@Service
@Profile("localci")
public class LocalCIDockerService {
```